src/components/Search.tsx

```diff
@@ -0,0 +1,77 @@
+import React, { useState, useEffect } from 'react';
+import { Autocomplete, Stack, Box, TextField, Typography } from '@mui/material';
+import { useSearchArticlesQuery } from '../queries/articles.query';
+import { ROUTES } from '../utils/static';
+
+const Search = () => {
+    const [searchParams, setSearchParams] = useState('');
+
+    const {
+        data: articles,
+        refetch,
+    } = useSearchArticlesQuery(searchParams);
+
+    const handleSearchParamsChange = (event: { preventDefault: () => void; }, value: React.SetStateAction<string>) => {
+        setSearchParams(value);
+    };
+
+    useEffect(() => {
+        refetch();
+    }, [searchParams]);
```

Comment:
you need to debounce this so its not spamming api requests.
Add debounce of 750ms